controller/oidc_auth/storage.go

```diff
@@ -894,15 +893,25 @@ func (s *HybridStorage) ValidateTokenExchangeRequest(_ context.Context, request
 		return fmt.Errorf("exchange subject type (%s) is not supported", proofType)
 	}
 
-	allowedScopes := []string{oidc.ScopeOpenID}
-
-	for _, scope := range request.GetScopes() {
-		if scope == oidc.ScopeOfflineAccess {
-			allowedScopes = append(allowedScopes, scope)
+	for _, aud := range request.GetAudience() {
+		if aud != common.ClaimAudienceOpenZiti && aud != common.ClaimLegacyNative {
+			return fmt.Errorf("invalid audience reqeusted [%s]", aud)
 		}
 	}
 
-	request.SetCurrentScopes(allowedScopes)
+	scopes := request.GetScopes()
+
+	if len(scopes) == 1 && scopes[0] == "" {
+		//no scopes supplied
+	} else {
+		for _, scope := range request.GetScopes() {
+			if scope != oidc.ScopeOfflineAccess && scope != oidc.ScopeOpenID {
```

Comment:
seems like this if could be refactored mebbe?

Reply:
Oh, the empty block. No that is an issuue.